vars/terraformDeployJob.groovy

```diff
@@ -0,0 +1,111 @@
+def call(Map config) {
+	library("tdr-jenkinslib")
+	def terraformWorkspace = config.stage == "mgmt" ? "default" : config.stage
+
+	pipeline {
+		agent {
+			label "master"
+		}
+		//Parameters section required for display in client Jenkins jobs GUI.
+		//Selected values are not accessible within the function and must be passed into the function via the config map
+		parameters {
+			choice(name: "STAGE", choices: ["intg", "staging", "prod"], description: "The environment you are deploying Terraform changes to")
+		}
+		stages {
+			stage("Run git secrets") {
+				steps {
+					script {
+						tdr.runGitSecrets(config.repo)
+					}
+				}
+			}
+			stage('Run Terraform build') {
+				agent {
+					ecs {
+						inheritFrom 'terraform'
+						taskrole "arn:aws:iam::${env.MANAGEMENT_ACCOUNT}:role/${config.taskRoleName}"
+					}
+				}
+				environment {
+					TF_VAR_tdr_account_number = tdr.getAccountNumberFromStage(config.stage)
+					//no-color option set for Terraform commands as Jenkins console unable to output the colour
+					//making output difficult to read
+					TF_CLI_ARGS = "-no-color"
+				}
+				stages {
+					stage('Set up Terraform workspace') {
+						steps {
+							dir("${config.terraformDirectoryPath}") {
+								echo 'Initializing Terraform...'
+								sh "git clone https://github.com/nationalarchives/tdr-terraform-modules.git"
+								sshagent(['github-jenkins']) {
+									sh("git clone git@github.com:nationalarchives/tdr-configurations.git")
+								}
+								sh 'terraform init'
+								//If Terraform workspace exists continue
+								sh "terraform workspace new ${terraformWorkspace} || true"
+								sh "terraform workspace select ${terraformWorkspace}"
+								sh 'terraform workspace list'
```

Comment:
Do you need to do workspace list here?

Reply:
No, it was just left there for legacy purposes. Will remove